docs/start/types.basics.md

```diff
@@ -0,0 +1,65 @@
+
+As a minimum, anything returned by the API, be it a `Vec<...>`, `Option<...>`, `Struct` or any normal type will always have the following methods -
+
+- `.eq(<other value>)` - checks for equality against the other value. In all cases, it will accept "like" values, i.e. in the case of a number you can pass a primitive (i.e. `1`), a hex value ( i.e. `0x01`) or even a `Unit8Array`
+- `toHex()` - returns a hex-base representation of the value, always prefixed by `0x`
+- `toJSON()` - returns a JSON-like representation of the value, this is generally use when calling `JSON.stringify(...)` on the value
+- `toString()` - returns a string representation, in some cases this performs additional encoding, i.e. for `Address` and `AccountId` it will encode to the ss58 address
+- `.toU8a()` - returns a `Uint8Array` representation of the encoded value (generally exactly as passed to the node, where values are SCALE encoded)
+
+Additionally, the following getters and utilities are available -
+
+- `.isEmpty` - `true` if the value is an all-empy value, i.e. `0` in for numbers, all-zero for Arrays (or anything `Uint8Array`)
+- `.hash` - a `Hash` (once again with all the methods above) that is a `blake2-256` of the contained value
+
+## Working with numbers
+
+All numbers wrap and extend an instance of [bn.js](https://github.com/indutny/bn.js/). This means that in addition  to the interfaces defined above, they have some additional methods -
+
+- `.toNumber()` - a JS number (limited to 2^53 - 1). This does mean that for large values, e.g. `Balance` (a `u128` extension), can cause overflows
+- `.add(...)`, `.sub(...)`, ... - all the base methods available on the `BN` object
+
+In cases where a `Compact` is returned, i.e. `Compact<Balance>`, the value is wrapped. This object should be `.unwrap()`-ed first to gain access to the underlying `Balance` object.
+
+## Working with structures
+
+All structures, a wrapping of an object containing a number of member variables, is an implementation of a standard JS `Map` object, so all the functions available on  a `Map` such as `.entries()` are available. Additionally it is decorated with actual getters for the fields.
+
+As an example, a `Header` will have getters for the `.parentHash`, `.number`, `.stateRoot`, `.extrinsicsRoot` and `.digest` fields. The same applies for all structures, as they are returned, each member will have an associated getter.
+
+## Working with enums
+
+Each enum has additional getters which are injected based on the fields wrapped. These take the form of `.is<Name>` and `.as<Name>` to allow you to check is the enum is a certain value or to retrieve the underlying value as a specific type.
+
+As an example, when an extrinsic is applied, the `Phase` enum has one of two states, `ApplyExtrinsic(u32)` or `Finalization`. In this case `.isApplyExtrinsic` would be `true` when an extrinsic is being applied, and `.asApplyExtrinsic` would return the phase value as a `u32`
```

Comment:
What would `.asApplyExtrinsic` return when the extrinsic is `Finalization`?

Reply:
It will throw.

Reply:
Actually needs some clarification here - on apply the u32 value is the index (in the block) of the extrinsic being applied

Reply:
Clarified.